File: projects/slick/lib/slick/commands/generate_model.rb

require "slick/command"

Slick::Command.register "generate-model" do

    arg "name"

    def run
        migration_dir_path = "#{project.root_path}/lib/#{project.name.snakeify}/models/";

        generate_file "#{migration_dir_path}/#{name.snakeify}.rb" do
            line
            line "Slick::Database::Row.register \"#{name.snakeify}\" do"
            indent do
                line
            end
            line "end"
            line
        end
    end

end

File: packages/sintra/lib/commands/generate_model.js

export default {
    async run(){

        const { extractArg, extractFields } = this.cliUtils;
        const name = this.inflector.snakeify(extractArg(''));
        if(name == ''){
            console.error('A model name must be given.');
            process.exit();
        }
        const fields = extractFields();
    
        const collectionName = this.inflector.camelize(this.inflector.pluralize(name));
        if(!await this.database[collectionName].exists){
            const denormalizedFields = fields.map(({ mandatory, name, type }) => {
                return `${ mandatory ? '^' : '' }${name}:${type}`
            });
            await this.runCommand('generate-migration', `create_${name}`, ...denormalizedFields, '--table', collectionName)
        }
    
        const { inProjectRootDir, generateFile, line, indent } = this.fsBuilder;
    
        await inProjectRootDir(async () => {
    
            await generateFile(`lib/models/_file_importer.js`, { skipIfExists: true }, () => {
                line();
                line(`export { Row as default } from 'sintra/database';`);
                line();
            });
    
            await generateFile(`lib/models/${name}.js`, () => {
                line();
                line(`export default {`);
                indent(() => {
                    line();
                });
                line('};');
                line();
            });
    
        });
    }
}
